#
# Since OpenSSH 10.2, we fail when the user name no longer matches;
# this check seems to fail with AD as it may return the username in all
# CAPS.
#
# Filed upstream as  https://bugzilla.mindrot.org/show_bug.cgi?id=3930
#
--- old/auth-pam.c
+++ new/auth-pam.c
@@ -335,7 +335,8 @@ check_pam_user(Authctxt *authctxt)
 
 	if (sshpam_initial_user == NULL)
 		fatal_f("internal error: sshpam_initial_user NULL");
-	if (strcmp(sshpam_initial_user, pam_user) != 0) {
+	/* For AD, we may get the username all in caps. */
+	if (strcasecmp(sshpam_initial_user, pam_user) != 0) {
 		error_f("PAM user \"%s\" does not match previous \"%s\"",
 		      pam_user, sshpam_initial_user);
 		return PAM_USER_UNKNOWN;
